examples/echo_server.hs

```diff
@@ -0,0 +1,18 @@
+{-# OPTIONS -XOverloadedStrings #-}
+
+import qualified Network.Freddy as Freddy
+import Data.ByteString.Lazy.Char8 (ByteString)
+
+processMessage :: Freddy.Request -> IO ()
+processMessage (Freddy.Request body replyWith failWith) = replyWith body
+
+main = do
+  fConnection <- Freddy.connect "127.0.0.1" "/" "guest" "guest"
```

Comment:
The significant part isn't just *monads* but the *IO monad* specifically, if I'm not mistaken